tests/fixtures/fixtures.py

```diff
@@ -29,6 +29,10 @@ def setUp(self):
         logger.add_output(NullOutput())
         deterministic.set_seed(1)
 
+        # initialize global singleton_pool for each test case
+        from garage.sampler import singleton_pool
+        singleton_pool.initialize(1)
```

Comment:
Does this help to fix any hanging bug?

Reply:
This [test](https://github.com/rlworkgroup/garage/issues/713#issuecomment-498942326) would not pass without it. The `test_batch_sampler()` is logger's problem. I'll take a deeper look to see if we can do something to resolve it. Essentially this helps with any hanging bugs due to the global `singleton_pool`, similar to the `snapshotter` one.